test/standalone/test-trace-gcloud.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+var common = require('../hooks/common.js');
+var nock = require('nock');
+var assert = require('assert');
+
+nock.disableNetConnect();
+process.env.GOOGLE_APPLICATION_CREDENTIALS =
+    __dirname + '/../fixtures/gcloud-credentials.json';
+
+describe('test-trace-gcloud', function() {
+  // This does a gcloud.datastore.get() request that makes a gRPC 'lookup' call.
+  // It attempts to authenticate using Google Auth by connecting to
+  // 'accounts.google.com:443/o/oauth2/token', but fails because of Nock.
+  // An auth error is returned, and a trace span for the gRPC call is created
+  // with an error.
+  it('should create gRPC spans', function(done) {
+    common.runInTransaction(function(endTransaction) {
+      var gcloud = require('../hooks/fixtures/gcloud0.36');
+      var ds = gcloud.datastore();
+      var key = ds.key(['bad', 'key']);
+      ds.get(key, function(err, entity) {
+        endTransaction();
+        assert(err);
+        assert.strictEqual(err.code, 401);
+        assert.strictEqual(err.message, 'Unauthorized');
+        var trace = common.getMatchingSpan(grpcPredicate);
+        assert(trace);
+        assert.notStrictEqual(trace.labels.argument.indexOf(
+            '"keys":[{"path":[{"kind":"bad","name":"key"}]}]'), -1);
+        assert(trace.labels.error);
```

Comment:
Let's unset the environment variable here just in case this gets run with other tests in the future.